File: src/main/java/jp/kentan/minecraft/neko_event/util/Log.java
package jp.kentan.minecraft.neko_event.util;

import org.bukkit.ChatColor;
import org.bukkit.Server;

import java.util.logging.Logger;

public class Log {

    private final static String PREFIX_INFO = ChatColor.translateAlternateColorCodes('&', "&7[NekoEvent &fINFO&7]&r ");
    private final static String PREFIX_WARN = ChatColor.translateAlternateColorCodes('&', "&7[NekoEvent &eWARN&7]&r ");
    private final static String PREFIX_ERROR = ChatColor.translateAlternateColorCodes('&', "&7[NekoEvent &4ERROR&7]&r ");

    private static Server sServer;
    private static Logger sLogger;

    public static void set(Server server, Logger logger){
        sServer = server;
        sLogger = logger;
    }

    public static void info(final String str){
        sLogger.info(str);
        sendMessageToOp(PREFIX_INFO + str);
    }

    public static void warn(final String str){
        sLogger.warning(str);
        sendMessageToOp(PREFIX_WARN + str);
    }

    public static void error(final String str){
        sLogger.warning(str);
        sendMessageToOp(PREFIX_ERROR + str);
    }

    private static void sendMessageToOp(final String message){
        sServer.getOnlinePlayers().forEach(p -> {
            if(p.isOp()){
                p.sendMessage(message);
            }
        });
    }
}

File: src/main/java/jp/kentan/minecraft/nekoevent/util/Log.kt
package jp.kentan.minecraft.nekoevent.util

import org.bukkit.Bukkit
import org.bukkit.ChatColor
import java.util.logging.Level
import java.util.logging.Logger

class Log {
    companion object {
        private val INFO  = ChatColor.translateAlternateColorCodes('&', "&7[Event &fINFO&7]&r ")
        private val WARN  = ChatColor.translateAlternateColorCodes('&', "&7[Event &eWARN&7]&r ")
        private val ERROR = ChatColor.translateAlternateColorCodes('&', "&7[Event &4ERROR&7]&r ")

        private val SERVER = Bukkit.getServer()

        lateinit var logger: Logger

        fun info(msg: String) {
            logger.log(Level.INFO, msg)
            sendOp(INFO + msg)
        }

        fun warn(msg: String) {
            logger.log(Level.WARNING, msg)
            sendOp(WARN + msg)
        }

        fun error(msg: String) {
            logger.log(Level.SEVERE, msg)
            sendOp(ERROR + msg)
        }

        fun error(exception: Exception) {
            exception.message?.let { error(it) }
        }

        private fun sendOp(msg: String) {
            SERVER.onlinePlayers.filter { it.isOp }.forEach { it.sendMessage(msg) }
        }
    }
}